### PR TITLE
[Merged by Bors] - feat: Function.End.one_def and mul_def

### PR DESCRIPTION
**Pull request by ChrisHughes24**
---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/GroupTheory/GroupAction/Defs.lean [hunk 1]
```diff
@@ -1128,6 +1128,12 @@ theorem Function.End.smul_def (f : Function.End α) (a : α) : f • a = f a :=
   rfl
 #align function.End.smul_def Function.End.smul_def
 
+theorem Function.End.mul_def (f g : Function.End α) : (f * g) = f ∘ g :=
```
Comment (eric-wieser) [inline Mathlib/GroupTheory/GroupAction/Defs.lean:1131]: This feels like defeq abuse to me; I'd be happier with something like
```suggestion
theorem Function.End.mul_def (f g : Function.End α) : (f * g) = toEnd (f ∘ g) :=
```

Comment (eric-wieser) [reply Mathlib/GroupTheory/GroupAction/Defs.lean:1131]: I'd be happy with a TODO comment instead, as long as these aren't simp lemmas

Comment (ChrisHughes24) [reply Mathlib/GroupTheory/GroupAction/Defs.lean:1131]: I'm not sure exactly what is TODO. There are a few ways `toEnd` could be done, we could completely forbid using `End` as a function without using a `FunLike` instance, or be less strict.

Comment (ChrisHughes24) [reply Mathlib/GroupTheory/GroupAction/Defs.lean:1132]: I added some comments

**Review by eric-wieser — APPROVED** [review]
Thanks!

bors merge

**Pull request closed without merging**